tests/robustness/model/deterministic.go

```diff
@@ -281,14 +317,20 @@ type EtcdCondition struct {
 }
 
 type EtcdOperation struct {
-	Type       OperationType
-	Key        string
-	WithPrefix bool
-	Limit      int64
-	Value      ValueOrHash
-	LeaseID    int64
+	Type OperationType
+	Key  string
+	RangeOptions
+	PutOptions
 }
 
+type OperationType string
+
+const (
+	RangeOperation  OperationType = "range-operation"
+	PutOperation    OperationType = "put-operation"
+	DeleteOperation OperationType = "delete-operation"
+)
```

Comment:
Can I think of the operation here is a similar alias of the following `RequestOp` used in the TxnRequest with one exception that `EtcdOperation` in robustness test was also used for robustness test watch event to patch the history and remove failed writes? 

https://github.com/etcd-io/etcd/blob/b4f49a55a5aba85a30cae5742a87989132a0f750/api/etcdserverpb/rpc.proto#L571-L580

Reply:
Yes, those are sub-requests within a transaction request, in model I just called them operations. We are missing TxnRequest as model doesn't support subtransactions currently (think it's overkill).